airflow/providers/hashicorp/hooks/vault.py

```diff
@@ -112,7 +112,7 @@ class VaultHook(BaseHook):
     """
 
     conn_name_attr = 'vault_conn_id'
-    default_conn_name = 'imap_default'
+    default_conn_name = 'vault_default'
```

Comment:
I think that this connection needs to be created on startup, with all of the other `<provider>_default` connections.

Interestingly, it looks like the `imap_default` connection isn't created there either. 

https://github.com/apache/airflow/blob/5980d2b05eee484256c634d5efae9410265c65e9/airflow/utils/db.py#L93-L113